src/services/middlewareServices/AuthenticationMiddlewareService.ts

```diff
@@ -10,63 +11,135 @@ import { AuthError } from "@errors/AuthError"
 
-const E2E_TEST_REPO = config.get("cypress.e2eTestRepo")
+export const E2E_TEST_REPO = config.get("cypress.e2eTestRepo")
+export const E2E_EMAIL_TEST_REPO = config.get("cypress.e2eEmailTestRepo")
 const E2E_TEST_SECRET = config.get("cypress.e2eTestSecret")
-const E2E_TEST_GH_TOKEN = config.get("cypress.e2eTestGithubToken")
-const E2E_TEST_USER = "e2e-test"
+
+export const E2E_TEST_GH_TOKEN = config.get("cypress.e2eTestGithubToken")
+export const E2E_TEST_USER = "e2e-test"
 const GENERAL_ACCESS_PATHS = [
   "/v1/sites",
   "/v1/auth/whoami",
   "/v2/sites",
   "/v2/auth/whoami",
 ]
 
-type VerifyAccessProps = SessionData & {
-  cookies: {
-    isomercms: string
-    isomercmsE2E?: string
-  }
+type UnverifiedSession =
+  | Partial<SessionData>
+  | { userInfo: Record<string, never> }
+  | { userInfo: SetOptional<SessionData["userInfo"], "email"> }
+
+export type VerifyAccessProps = UnverifiedSession & {
+  // NOTE: Either both properties are present on the cookie
+  // or none are present.
+  // We disallow having 1 or the other.
+  cookies: RequireAllOrNone<
+    {
+      isomercmsE2E: string
+      e2eUserType: string
+    },
+    "e2eUserType" | "isomercmsE2E"
+  >
   url: string
 }
 
+type TestUserTypes = "Email admin" | "Email collaborator" | "Github user"
+
+// NOTE: Precondition to use this function is that the user type is valid.
+const getUserType = (userType: string): TestUserTypes => {
+  if (userType === "Email admin") return "Email admin"
+  if (userType === "Email collaborator") return "Email collaborator"
+  if (userType === "Github user") return "Github user"
+  throw new Error(`Invalid user type: ${userType}`)
+}
+
+const extractE2eUserInfo = (userType: TestUserTypes): SessionDataProps => {
+  switch (userType) {
+    case "Email admin":
+      return {
+        isomerUserId: E2E_EMAIL_ADMIN_ISOMER_ID,
+        email: E2E_TEST_EMAIL,
+      }
+    case "Email collaborator":
+      return {
+        isomerUserId: E2E_EMAIL_COLLAB_ISOMER_ID,
+        email: E2E_TEST_EMAIL,
+      }
+    case "Github user":
+      return {
+        accessToken: E2E_TEST_GH_TOKEN,
+        githubId: E2E_TEST_USER,
+        isomerUserId: E2E_ISOMER_ID,
+        email: E2E_TEST_EMAIL,
+      }
+    default: {
+      const missingUserType: never = userType
+      throw new Error(`Missing user type: ${missingUserType}`)
+    }
+  }
+}
+
 export default class AuthenticationMiddlewareService {
-  verifyE2E({ cookies, url }: Omit<VerifyAccessProps, "userInfo">) {
-    const { isomercmsE2E } = cookies
+  private verifyE2E({
+    cookies,
+    url,
+  }: Omit<VerifyAccessProps, "userInfo">): TestUserTypes | false {
+    const { isomercmsE2E, e2eUserType } = cookies
     const urlTokens = url.split("/") // urls take the form "/v1/sites/<repo>/<path>""
 
-    if (!isomercmsE2E) return false
+    // NOTE: If the cookie is not set, this is an actual user.
+    if (!isomercmsE2E || !e2eUserType) return false
```

Comment:
Just to check my understanding here ya, actually this can be simpified to 
if (!isomercmsE2E) return false;
right, since `e2eUserType` will be present as long as `e2eUserType` is present?

Reply:
that's true at _compile_ time but because we don't validate it, we have to do this check for both.